src/main/java/com/incognia/api/IncogniaAPI.java

```diff
@@ -336,7 +345,9 @@ public void registerFeedback(
             .signupId(identifiers.getSignupId())
             .externalId(identifiers.getExternalId())
             .build();
-    tokenAwareNetworkingClient.doPost("api/v2/feedbacks", requestBody);
+    Map<String, String> queryParameters = new HashMap<>();
+    queryParameters.put(DRY_RUN_PARAMETER, dryRun.toString());
```

Comment:
I think you'll run into a nullpointer if the user sends null in the dryRun parameter here, since you'll try to call a method from a null reference.
Maybe before line 348 we can do a checking of null and set to False if the dryRun parameter is null?
`
dryRun = Objects.isNull(dryRun) ? false : dryRun;
`

Also, i think we can separate the block starting in line 337 from this new one, so the method becomes easier to read since we have well defined blocks clustered together